Rematch/Addon.lua

```diff
@@ -18,33 +18,33 @@ function Addon:OnInitialize()
 end
 
 function Addon:GetCurrentKey()
-    return RematchSettings.loadedTeam
+    return Rematch.settings.currentTeamID
```

Comment:
open question/todo: maintain rematch 4 compatibility for the transition period?

Reply:
Making use of that `Version.lua` file/class to load the hooks/events depending on the version of rematch is the way to go there. 

But the real issue is loading the scripts from the database. The key in the database key is the profile name, and it appears the script's for that plugin are loaded before the plugin is even enabled. 

Maybe we can use a separate plugin for Rematch5, and check which version of Rematch has loaded when the both plugins are enabled?

Reply:
Once `RematchPlugin:OnEnabled()` is called, rematch is loaded. The bad part is that at some times `RematchPlugin:*` is called without `:OnEnabled()` ever being called. I think I got all those cases covered, making it now possible to use version everywhere with local conditions and thus little duplication.